components/brave_shields/browser/domain_block_controller_client.cc

```diff
@@ -55,8 +56,8 @@ void DomainBlockControllerClient::Proceed() {
   tab_storage->SetIsProceeding(true);
   if (dont_warn_again_) {
     std::string custom_filters =
-        ad_block_custom_filters_service_->GetCustomFilters();
-    ad_block_custom_filters_service_->UpdateCustomFilters(
+        ad_block_custom_filters_source_provider_->GetCustomFilters();
+    ad_block_custom_filters_source_provider_->UpdateCustomFilters(
```

Comment:
we're actually modifying the filter to disable the rule? That's definitely not great, but a separate issue from this PR. Can you open a ticket for that? In any case these methods should be called on the service, not the source provider and `DomainBlockControllerClient` shouldn't have to know anything about the rules format. We should just be calling `ad_block_custom_filters_service_->DisableRule(request_url_.host())` or something along those lines

Reply:
Filed a followup at https://github.com/brave/brave-browser/issues/19863

Reply:
but both this class and DomainBlockNavigationThrottle should only ever talk to the service and shouldn't know anything about the underlying source provider so you should pass the service here and add methods to pass through to the source provider

Reply:
+1 !